solarforecastarbiter/metrics/tests/test_preprocessing.py

```diff
@@ -506,8 +506,8 @@ def test_align_prob_constant_value(
             # only first all-nan interval is discarded
             pd.Series([7.0, 10.5, 14.5], index=FOUR_HOUR_SERIES.index[1:]),
             (('ISNAN', 5), ('TOTAL DISCARD BEFORE RESAMPLE', 5),
-             ('NIGHTTIME OR USER FLAGGED OR ISNAN', 0, False),
-             ('TOTAL DISCARD AFTER RESAMPLE', 0, False))
```

Comment:
Maybe it has to do with the fact that the four 15 minute obs in the hour were all nan values instead of all some flag. But I still don't see how that works out.

Reply:
I think this is found in `_calc_discard_before_resample` here: https://github.com/SolarArbiter/solarforecastarbiter-core/blob/23122548ec968ea4ec59f4877cc65f6e23994d4e/solarforecastarbiter/metrics/preprocessing.py#L314-L330

ISNAN is *always* discarded before resample, so we would expect that first interval to disappear in the output. However, the `to_discard_after_resample` is calculated based on the index of the `to_discard_before_resample` series https://github.com/SolarArbiter/solarforecastarbiter-core/blob/23122548ec968ea4ec59f4877cc65f6e23994d4e/solarforecastarbiter/metrics/preprocessing.py#L374-L375
which will then detect that it should discard that interval, because the ISNAN flag was present for 100% of the data.

Reply:
Thanks, now I get it. We should have expected the interval to be dropped because it was all NAN. And the fact that it was dropped while `_calc_discard_after_resample` returned flags with `count=0` should have been a warning that something was wrong.